Assignment5/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+
+namespace GenericsHomework
+{
+    public class Node<T>
+    {
+        private T? _Value;
+        private Node<T>? _Next;
+
+        public T Value { get => _Value!; private  set => _Value = value;}
+
+        public Node<T> Next
+        {
+            get => _Next!;
+            set
+            {
+                value._Next = this;
+                _Next = value ?? throw new ArgumentNullException(nameof(value));
+            }
+        }
+
+        public Node(T value)
+        {
+            Value = value;
+            Next = this;

```

Comment:
```suggestion
            _Value = value;
            _Next = this;
```
Use the private variables here to avoid marking them as nullable.